src/WpfMath/Atoms/Atom.cs

```diff
@@ -1,19 +1,26 @@
+using System.Collections.Generic;
 using WpfMath.Boxes;
-
 namespace WpfMath.Atoms
 {
     // Atom (smallest unit) of TexFormula.
     internal abstract class Atom
     {
-        protected Atom(SourceSpan source, TexAtomType type = TexAtomType.Ordinary)
+        protected Atom(SourceSpan source, IEnumerable<Atom> children, TexAtomType type = TexAtomType.Ordinary)
         {
             this.Source = source;
             this.Type = type;
+            this.Children = children;
+        }
+
+        protected Atom(SourceSpan source, TexAtomType type = TexAtomType.Ordinary): this(source, new List<Atom>(), type)
+        {
         }
 
         public TexAtomType Type { get; }
 
-        public SourceSpan Source { get; }
+        public SourceSpan Source { get; set; }
+
+        public IEnumerable<Atom> Children { get; protected set; }
```

Comment:
I also thought that more atoms deserve special `Children` processing: every atom that has some children kept in its properties should override/provide the `Children` property. Otherwise, it won't work well.

Currently, you've only implemented it in the `RowAtom`, this isn't enough.